pkg/image/registry/image/strategy.go

```diff
@@ -31,7 +32,13 @@ func (imageStrategy) NamespaceScoped() bool {
 }
 
 // PrepareForCreate clears fields that are not allowed to be set by end users on creation.
+// It extracts the latest information from the manifest (if available) and sets that onto the object.
```

Comment:
Yes, I have a follow up that when I fix upgrade that will refetch and update any image that doesn't have the manifest in it.